metsim/io.py

```diff
@@ -146,18 +146,18 @@ def read_netcdf(data_handle, domain=None, iter_dims=['lat', 'lon'],
     ds = xr.open_dataset(data_handle)
 
     if var_dict is not None:
+        var_list = list(var_dict.keys())
+        ds = ds[var_list]
         ds.rename(var_dict, inplace=True)
 
-    if start is not None and stop is not None:
+    if start is not None or stop is not None:
```

Comment:
Doesn't this have the chance to break? For example, if `start=None` and `stop!=None` we will end up with (in the line below) `ds = ds.sel(time=slice(None, stop))`.